src/ui/wxWidgets/DbSelectionPanel.cpp

```diff
@@ -137,3 +169,39 @@ void DbSelectionPanel::OnFilePicked(wxFileDirPickerEvent& WXUNUSED(event))
   if ( !wxDynamicCast(FindFocus(), wxTextCtrl) )
     m_sc->SelectCombinationText();
 }
+
+#ifndef NO_YUBI
+/*!
+ * wxEVT_COMMAND_BUTTON_CLICKED event handler for ID_YUBIBTN
+ */
+
+void DbSelectionPanel::OnYubibtnClick(wxCommandEvent& WXUNUSED(event))
+{
+  m_sc->AllowEmptyCombinationOnce();  // Allow blank password when Yubi's used
+
+  if (Validate() && TransferDataFromWindow()) {
+    StringX response;
+    bool oldYubiChallenge = ::wxGetKeyState(WXK_SHIFT); // for pre-0.94 databases
+    if (PerformChallengeResponse(this, m_combination, response, oldYubiChallenge)) {
+      m_combination = response;
+
+      /* Compare Dialog
+      GetParent()->GetEventHandler()->AddPendingEvent(wxCommandEvent(wxEVT_BUTTON, ID_BTN_COMPARE));
+      */
+
+      /* Merge Dialog
+      GetParent()->GetEventHandler()->AddPendingEvent(wxCommandEvent(wxEVT_BUTTON, wxID_OK));
+      */
+
```

Comment:
DbSelectionPanel is used as an embedded control in the Merge and Compare dialog. I'm not sure what should happen after YubiMixin::PerformChallengeResponse in DbSelectionPanel::OnYubibtnClick was successful in the case of the merge or in the comparison dialog. Should the "OK" button click event be triggered in the Merge dialog and the "Compare" button click event in the  Compare dialog?

Reply:
Semantically, OnYubibtnClick() should set the m_combination field and simulate the OK / Compare button, so I think this part is OK.
Only problem I see is that the CheckPasskey() in DoValidation() should check the m_combination value set in line 106 above, and *not* what's xferred from the control in the call to wxWindow::Validate() in DoValidation.